### PR TITLE
Support prefix search for API key names

### LINKED ISSUE #57638: Improve search filtering for GET API keys API
The GET API keys API can take a few query parameters to filter the response. For an example, the `name` parameter can be used to find API keys that have the given name. Currently all these parameters support only exact match and this may not work well where there are a large number of API keys. This could be improved by, e.g. supporting wildcard for the `name` fields.

### PR DESCRIPTION
**Pull request by ywangd**
This PR adds minimum support for prefix search of API Key name. It only touches API key name and leave all other query parameters, e.g. realm name, username unchanged. It's easy to add them, but they may not be of much value since for the scenario of large number of API keys, we do not expect as many realm names or usernames compared to API key names. 

I also think we haven't discussed about the possibilities thoroughly enough. So I am adding minimum support since it is easy to give out but hard to take back.

Resolves: #57638

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java [hunk 1]
```diff
@@ -532,13 +532,21 @@ public void testGetApiKeysForApiKeyId() throws InterruptedException, ExecutionEx
...
         GetApiKeyResponse response = listener.get();
         verifyGetResponse(1, responses, response, Collections.singleton(responses.get(0).getId()), null);
+
+        PlainActionFuture<GetApiKeyResponse> listener2 = new PlainActionFuture<>();
+        client.execute(GetApiKeyAction.INSTANCE, GetApiKeyRequest.usingApiKeyName("test-key*", false), listener2);
+        verifyGetResponse(3, responses, listener2.get(), responses.stream().map(o -> o.getId()).collect(Collectors.toSet()), null);
+
+        PlainActionFuture<GetApiKeyResponse> listener3 = new PlainActionFuture<>();
+        client.execute(GetApiKeyAction.INSTANCE, GetApiKeyRequest.usingApiKeyName("*", false), listener2);
+        verifyGetResponse(3, responses, listener2.get(), responses.stream().map(o -> o.getId()).collect(Collectors.toSet()), null);
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java:549]: Can we have a non-match wildcard as well?

```
        PlainActionFuture<GetApiKeyResponse> listener4 = new PlainActionFuture<>();
        client.execute(GetApiKeyAction.INSTANCE, GetApiKeyRequest.usingApiKeyName("not-test-key*", false), listener4);
        assertThat(listener4.get().getApiKeyInfos(), arrayWithSize(0));
```

Ideally a subset match (find 2 out of the 3 keys) would be nice, but the test infrastructure in this class doesn't make that easy.

Comment (ywangd) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java]: Thanks. Added tests for both "non-matching wildcard" and "subset match".

**Review by tvernum — APPROVED** [review]
LGTM

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-unix

**Review by albertzaharovits — APPROVED** [review]

**Pull request merged**